### PR TITLE
A slightly more configurable footer

### PR DESCRIPTION
**Pull request by pivotal-gordon-krull**
I have added configuration for `is-fluid` classname to the Footer sections via `isFluid` prop (similar to other components like `Container` and `MainNav`).

Included as well are some quality of life updates:
- Converted `Footer` to a React functional component
- Added PropTypes for existing props
- Added example Fluid Footer to the `Footer` stories in Storybook

### CODE REVIEW

File: src/components/Footer.js [hunk 1]
```diff
@@ -1,112 +1,127 @@
...
+        </div>
+      </div>
+    </footer>
+  );
+};
+
+Footer.propTypes = {
+  title: PropTypes.string,
+  date: PropTypes.string,
+  links: PropTypes.objectOf({
```
Comment (weijunyu) [inline src/components/Footer.js:118]: This currently throws an error with the examples, I believe the proptype should be `shapeOf` instead

File: src/components/Footer.js [hunk 2]
```diff
@@ -1,112 +1,127 @@
...
+    privacy: PropTypes.string,
+    termsOfUse: PropTypes.string,
+    contact: PropTypes.string,
+    feedback: PropTypes.string
+  }),
+  isFluid: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
```
Comment (weijunyu) [inline src/components/Footer.js:124]: I think we are only using bool here :)

**Review by weijunyu — CHANGES_REQUESTED** [review]

File: src/components/Footer.js (hunk 2 above)
Comment (pivotal-gordon-krull) [reply src/components/Footer.js]: I have updated to only use bool. I wasn't entirely sure and was using `Container` as a reference here which has both.

File: src/components/Footer.js (hunk 1 above)
Comment (pivotal-gordon-krull) [reply src/components/Footer.js]: Fixed.

**Pull request merged**